src/core/ext/transport/chaotic_good/client_transport.cc

```diff
@@ -113,6 +123,73 @@ ClientTransport::ClientTransport(
       [](absl::Status status) {
         GPR_ASSERT(status.code() == absl::StatusCode::kCancelled ||
                    status.code() == absl::StatusCode::kInternal);
+        // TODO(ladynana): handle the promise endpoint write failures with
+        // outgoing_frames.close() once available.
+      });
+  auto read_loop = Loop([this] {
+    return TrySeq(
+        // Read frame header from control endpoint.
+        this->control_endpoint_->Read(FrameHeader::frame_header_size_),
+        // Read different parts of the server frame from control/data endpoints
+        // based on frame header.
+        [this](SliceBuffer read_buffer) mutable {
+          frame_header_ = std::make_shared<FrameHeader>(
+              FrameHeader::Parse(
+                  reinterpret_cast<const uint8_t*>(GRPC_SLICE_START_PTR(
+                      read_buffer.c_slice_buffer()->slices[0])))
```

Comment:
Why are we assuming here that the whole frame header will be in the first slice i.e slice at index 0 ? Is it possible for the promise endpoint to return the frame header split across many slices ?